session-9/src/hooks/useGithubMembersByCity.js

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useState } from "react";
+
+const useGitHubMembersByCity = (selectedCity) => {
+  const [loading, setLoading] = useState(false);
+  const [data, setData] = useState([]);
+  useEffect(() => {
+    console.log("Hello");
+    (async () => {
+      setLoading(true);
```

Comment:
use always try-catch inside the async function, or what is the use of the async function then?

Reply:
Yes it's right always good to use try/catch for handling server issues. But I guess still this function need to async as it need to wait for promise to resolve. wrapping this function with try/catch.

Reply:
Don't wrap it inside try-catch, use try-catch inside it.

```
(async () => {
      try{
      setLoading(true);
      const response = await fetch(
        `https://api.github.com/search/users?q=location:${selectedCity}`
      );
      const data = await response.json();
      setData(data);
      } catch (error){
      console.log(error)
      }finally{
      setLoading(false);
      })()
```

Reply:
Yes right, But still it does bubble out right till it gets caught.😀

Reply:
You didn't get it.